CONTRIBUTING.md

```diff
@@ -11,95 +11,156 @@ open an issue and discuss the feature with us. Sending a PR without discussion
 might result in a rejected PR, because we might be taking the repository in a
 different direction.
 
-## Building from source
+## Setting up the Workspace
 
-We recommend you use our prebuilt Docker image to start your development work
-using either VS Code or a local container:
+Please follow the steps below in order:
 
-### Visual Studio Code Dev Container
+### Prerequisite
 
-* Create an empty directory for your workspace on your development host. These
-  instructions assume you are using a remote host and are connecting to it over
-  SSH.
+To work on PyTorch/XLA, you'll need a powerful Linux machine with plenty of
+CPUs and RAM. Make sure you have `git` and `docker` installed on this machine.
+If you don't, follow https://github.com/git-guides/install-git and
+https://docs.docker.com/engine/install/ to install them.
 
-* Clone PyTorch, TorchVision, and PyTorch/XLA into your workspace directory:
+### Forking the Git Repos
 
-```bash
-  git clone --recursive --depth=1 https://github.com/pytorch/pytorch.git
+In order to create PRs later, we need to first fork the Git repos we'll be
+working with:
 
-  # Install TorchVision if you need to run tests that involve vision modules
-  git clone --recursive --depth=1 https://github.com/pytorch/vision.git
+1.  Go to https://github.com/pytorch/pytorch and fork it as `pytorch`.
+1.  Go to https://github.com/pytorch/vision and fork it as `vision`.
+1.  Go to https://github.com/pytorch/xla and fork it as `pytorch-xla`. Note
+    the change of project name: we want to avoid confusion with the OpenXLA
+    project that PyTorch/XLA depends on, which is also named `xla`.
 
-  # Clone with HTTPS if you use a GitHub personal access token
-  git clone https://github.com/pytorch/xla.git pytorch/xla
+### Cloning the Forked Repos
 
-  # Or clone with SSH if you prefer:
-  git clone git@github.com:pytorch/xla.git pytorch/xla
-```
+Next, we need to clone the forked repos locally so that we can make changes.
 
-* Create links to VS Code configuration files in your workspace directory:
+On your Linuc machine, decide a directory as your workspace. Make sure that
+this directory and all of its ancestors are publically readable. Then run
+the following commands on this machine:
 
 ```bash
-  ln -s pytorch/xla/.devcontainer/ .devcontainer
-  ln -s pytorch/xla/contrib/vscode/ .vscode
-  ln -s pytorch/xla/.style.yapf .style.yapf
-  ln -s pytorch/xla/.clang-format .clang-format
+# Make sure that all new files and directories are publically readable.
+# Otherwise you may have permission issues when building the code in bazel's
+# sandbox mode.
+umask 022
+
+# Create the workspace directory if you haven't.
+export WORKSPACE_DIR=<absolute-path-to-your-workspace>
+mkdir -p $WORKSPACE_DIR
+
+# Clone the repos.
+cd $WORKSPACE_DIR
+git clone --recursive git@github.com:<your-github-user-name>/pytorch.git
```

Comment:
I would just clone from pytorch/pytorch.git

Reply:
See my reply in the other thread. Trying to provide a consolidated approach here, even though it may not be optimal.